src/cli/commands/test/mod.rs

```diff
@@ -0,0 +1,169 @@
+		proxies::{exec_scopes_proxy::ExecutionScopesProxy, vm_proxy::VMProxy},
+	},
+	serde::deserialize_program::ApTracking,
+	vm::{errors::{vm_errors::VirtualMachineError}},
+};
+use clap::{Args, ValueHint};
+use log::error;
+use serde::Serialize;
+
+use super::CommandExecution;
+use super::list::{path_is_valid_directory, ListArgs};
+
+
+use crate::cli::commands::execute::compile::compile;
+
+#[derive(Args, Debug)]
+pub struct TestArgs {
+	/// Path to a cairo program
+	#[clap(short, long, value_hint=ValueHint::FilePath, value_parser=path_is_valid_directory)]
+	pub program: PathBuf,
+}
+
+// fn is_cairo(path: &str) -> Result<PathBuf, String> {
+// 	let path = PathBuf::from(path);
+// 	if path.exists() && path.is_file() {
+// 		match path.extension() {
+// 			Some(ext) if ext == "cairo" => Ok(path),
+// 			_ => Err(format!("\"{}\" is not a cairo file", path.display())),
+// 		}
+// 	} else {
+// 		Err(format!("\"{}\" is not a valid file", path.display()))
+// 	}
+// }
+
+pub fn get_test_functions(compiled_path: &PathBuf) -> Result<Vec<String>, String> {
+	let re = Regex::new(r"__main__.(test_\w+)").map_err(|op| format!("Regex failed"))?;
+	let data = fs::read_to_string(compiled_path).map_err(|op| format!("File does not exist"))?;
+	let json = serde_json::from_str::<Value>(&data).map_err(|op|format!("Compile fails: {}", compiled_path.display()))?;
+	let mut test_functions = Vec::new();
+	// if json is empty, throw error
+	if json.is_null() {
+		return Err(format!("Json file is empty"));
+	}
+	for (key, value) in json["identifiers"].as_object().unwrap() {
+		if re.is_match(key) && value["type"] == "function" {	
+			for capture in re.captures_iter(key) {
+				test_functions.push(capture[1].to_string());
+			}
+		}
+	}
+	Ok(test_functions)
+}
+
+/// Execute command output
+#[derive(Debug, Serialize)]
+pub struct TestOutput(Vec<u8>);
+
+impl Write for TestOutput {
+	fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+		self.0.write(buf)
+	}
+
+	fn flush(&mut self) -> std::io::Result<()> {
+		self.0.flush()
+	}
+}
+
+
+
+impl Display for TestOutput {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(
+			f,
+			"{}",
+			from_utf8(&self.0).map_err(|e| {
+				error!("failed to format the execution output due to invalid utf8 encodig: {e}");
+				std::fmt::Error
+			})?
+		)
+	}
+}
+
+
+
+impl CommandExecution<TestOutput> for TestArgs {
+	fn exec(&self) -> Result<TestOutput, String> {
+		let hint = HintFunc(Box::new(greater_than_hint));
+		let mut hint_processor = BuiltinHintProcessor::new_empty();
+		hint_processor.add_hint(String::from("print(ids.a > ids.b)"), hint);
+
+		let result = ListArgs {
+			root: PathBuf::from(&self.program),
+		}.exec();
+		
+		let mut output = TestOutput(vec![]);
+		for file in result?.files {
+			let compiled_program_path = compile(&file)?;
+			let test_functions = get_test_functions(&compiled_program_path);
+
+			if test_functions.is_ok() && test_functions.as_ref().unwrap().is_empty() {
```

Comment:
This can go in the `ok` branch of your latter `match` :
```rust
			let test_functions = match test_functions {
				Ok(res) => res, // HERE
				Err(_) => {
					eprintln!("Compilation fails: {}", file.display());
					continue;
				}
			};
```

Reply:
i moved the error handling inside `list_test_entrypoints` instead

what do you think about it?

Reply:
reverted!